### PR TITLE
Use Redux in React Components

### PR DESCRIPTION
**Pull request by Mike111222**
I have done the following changes and I want to implement these:

- Wrap App with the Provider component from react-redux
- Pass configured store into the Components
- Create an initialState variable for booksSlice.
- Display books, received from the slice, in a reusable component
- Dispatch actions using useDispatch.
- Add an event handler that adds a book to the books array (with attributes id, title, and author)
- Add an event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-      books: [...state.books, action.payload],
-    }),
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.item_id !== action.payload);
+    },
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
   },
 });
```
Comment (AdnanOlarmmi) [inline src/redux/books/booksSlice.js]: - [ ] [OPTIONAL] One of the backbone and peculiarities of redux is not to mutate the store directly but rather using pure functions to update your store,

To make the Redux slice pure, you should avoid directly mutating the state inside the reducer functions. Here's how you can modify your code to make it pure:

```
const bookSlice = createSlice({
  name: 'book',
  initialState,
  reducers: {
    addBook: (state, action) => {
      return {
        ...state,
        books: [...state.books, action.payload],
      };
    },
    removeBook: (state, action) => {
      const itemId = action.payload;
      return {
        ...state,
        books: state.books.filter((book) => book.item_id !== itemId),
      };
    },
  },
});
```

File: src/components/BookCard.js [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
+          <li><button type="button">Comments</button></li>
+          <li><button type="button" onClick={() => dispatch(removeBook(book.item_id))}>Remove</button></li>
+          <li><button type="button">Edit</button></li>
+        </ul>
+      </div>
+      <div className="bookChart">
+        <div>Chart</div>
+        <div>
+          <p>progres</p>
+          <p>COMPLETED</p>
```
Comment (AdnanOlarmmi) [inline src/components/BookCard.js]: - [ ] According to the requirement, though it is not mentioned directly, the BookCard or Book is supposed to be a child of BooksList meaning BookCard and BookList should have a parent-child-relationship.

Books.js or BookCard.js
```
const Book = (props) => {
  const { book } = props;
  const { item_id, title, author, category } = book;

  return (
    <li className="book flex-row">
      <div className="bookInfo">
        <div className="flex-column">
          <span className="bookCategory">{category}</span>
          <span className="bookTitle">{title}</span>
          <span className="bookAuthor">{author}</span>
        </div>
      </div>
    </li>
  );
};
```

Then Books.js or BookList.js
will be something like
```
import BookCard.js

const Books = () => {

//dummy data
const books = [
{
author: 'you',
name: 'bookname'
}
]

  return (
    <main>
      <ul className="booksContainer flex-column">
        {books.map((book) => (<Book key={book.item_id} book={book} />))}
      </ul>
    </main>
  );
};
```

File: src/components/AddBook.js [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+          value={values.author || ''}
+          type="text"
+          name="author"
+          placeholder="author"
+          onChange={handleChange}
+        />
+
+        <button type="submit" onClick={handleSubmit}>Add Book</button>
+      </form>
+    </section>
```
Comment (AdnanOlarmmi) [inline src/components/AddBook.js:46]: - [ ] [OPTIONAL] You can also add category as a drop down input field.
253909755-cc4aa165-db0b-4d8a-9c20-ee6bf947db5f

**Review by AdnanOlarmmi — CHANGES_REQUESTED** [review]
Hi @Mike111222, 

This is Adnan Is-haq, your beloved code reviewer,

You've done a very good job so far, but a few things still need to be worked on before moving on to the next project, but you're so close!

You're Almost there, funny one eyed alien

## Highlights
- Clear and organized store design.
- Well-written reducer functions.
- Effective use of action creators.
- Efficient use of selectors.
- Well-connected components.
- Clear and well-organized code.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (Mike111222) [reply src/redux/books/booksSlice.js]: I have done the changes you mentioned, thanks for the kind feedback.

File: src/components/BookCard.js (hunk 2 above)
Comment (Mike111222) [reply src/components/BookCard.js]: I have done the changes you mentioned, thanks for the kind feedback. Thanks

**Review by ghost — APPROVED** [review]
Hi @Mike111222 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**